app/models/user.rb

```diff
@@ -442,8 +445,25 @@ def banned?
     !ban_text.empty?
   end
 
+  def closed?
+    closed_at.present?
+  end
+
+  def close_and_anonymise
+    sha = Digest::SHA1.hexdigest(rand.to_s)
+    update(
```

Comment:
We'll need to create the censor rule before we anonymise, otherwise we'll lose the original name that we need to redact.